Convert into pipenv

vrutkovs opened the pull request:
Use `pipenv` to manage dependencies, add flake8 linting via Github Actions and add Source2Image-based Dockerfile

wking commented on .gitignore:
```diff
@@ -3,3 +3,5 @@ __pycache__/
 *.egg-info/
 dist/
+cluster_support_bot.egg-info/
+.vscode/
```
This is an editor thing, right?  Editor gitignore entries should go in `$XDG_CONFIG_HOME/git/ignore` or similar user-scoped config file, per [`gitignore(1)`][1].

[1]: https://git-scm.com/docs/gitignore

vrutkovs replied:
Good catch, fixed

wking commented on cluster_support_bot/messages.py:
```diff
@@ -52,18 +35,40 @@ def print_help(self, file=None):
+    return web_client.chat_postMessage(channel=channel, thread_ts=thread, text=message)
+
+
+def handle_parse_args_error(payload, error):
```
It's hard for me to work out how much of this content is moved, vs. reformatted, vs.new, etc.  Can we break this down into more commits, so I can quickly skim over any moves/reformats, and more carefully review any actual functionality changes?  I'm also fine with "I've tried this, it works, don't worry about it, and just merge" ;)

vrutkovs replied:
>Can we break this down into more commits, so I can quickly skim over any moves/reformats, and more carefully review any actual functionality changes?

No functionality changes, most of this is done to allow proper package build/install along with "this needs to be moved out so that unit tests would work". Not sure if I can split this even more

vrutkovs replied:
Broke this apart is several commits - __init__.py with main loop and flake8 fixes

Commit:
Use pipenv to install dependecies

Instead of vague requirements.txt we should pin to particular versions of packages

Commit:
Use flake8 as a linter, fix lint errors

Commit:
Add unit tests

Commit:
Add __init__ file

This is an app entrypoint, containing asyncio webserver which calls `messages._handle_message` for every received Slack message. It also initialized Hydra client and passes it to messages class.
The app initializes Prometheus counters, which are used by messages class

Commit:
Read env vars when _query is called

This file is not imported by default now, so it needs to read env vars when _query is called

Commit:
handle_uuid_mention: return UUID and exit early if its not present
